daisy_workflows/build-publish/rhui/health_checks/health_check.py

```diff
@@ -0,0 +1,86 @@
+#  Copyright 2022 Google Inc. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http:#www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Health check for RHUI infrastructure nodes."""
+
+import argparse
+import logging
+import os
+import sys
+import typing
+
+
+def nfs_is_read_mounted(logger: logging.Logger):
+  nfs_mount = "/var/lib/rhui/remote_share"
+  logger.info("checking that %s is mounted", nfs_mount)
+  os.statvfs(nfs_mount)
+
+
+def nfs_is_write_mounted(logger: logging.Logger):
+  nfs_mount = "/var/lib/rhui/remote_share"
+  logger.info("checking that %s is writable", nfs_mount)
+  stat = os.statvfs(nfs_mount)
+  if bool(stat.f_flag & os.ST_RDONLY):
+    raise EnvironmentError("%s is read only" % nfs_mount)
```

Comment:
this checks the *permission of the mountpoint*, not the mount status. read-only vs read-write (which i assume is what you want to check). really, the simplest check is to first ensure it is mounted (can layer the other function for that), then write and delete a temp file. the tempfile package has context objects that do this automatically

Reply:
I like the tempfile idea. In the latest version, if I mount NFS with `mount -o ro`, then the following error is generated:

```
mount_is_writable - INFO - checking that /mnt/nfs is writable
mount_is_writable - ERROR - [Errno 30] Read-only file system: '/mnt/nfs/tmpdi_qfs_d'
```